inception-search-mtas/src/test/java/de/tudarmstadt/ukp/inception/search/index/mtas/MtasDocumentIndexTest.java

```diff
@@ -0,0 +1,557 @@
+                    log.info("**************** Index is NOT valid");
+                    uploadWaiter.assertTrue(false);
+                }
+            }
+            catch (InterruptedException e) {
+                e.printStackTrace();
+            }
+
+            uploadWaiter.resume();
+        }).start();
+
+        // Wait for thread to complete
+        uploadWaiter.await();
+
+        Waiter annotationWaiter = new Waiter();
+
+        // Start thread to annotate the document
+        new Thread(() -> {
+            try {
+                ClassLoader classLoader = getClass().getClassLoader();
+                File annotatedFile = new File(classLoader.getResource("galicia.conll").getFile());
+
+                JCas annotationCas = importExportService.importCasFromFile(annotatedFile, project,
+                        "conll2002");
+
+                AnnotationDocument annotationDocument = documentService
+                        .createOrGetAnnotationDocument(sourceDocument, user);
+
+                documentService.writeAnnotationCas(annotationCas, annotationDocument, false);
+            }
+            catch (IOException | UIMAException e) {
+                e.printStackTrace();
+            }
+
+            try {
+                // Wait some time so that the document can be indexed
+                Thread.sleep(WAIT_TIME);
+
+                // Test if the index has been created
+                if (searchService.isIndexValid(project)) {
+                    log.info("**************** Index is valid");
+                    uploadWaiter.assertTrue(true);
+                }
+                else {
+                    log.info("**************** Index is NOT valid");
+                    annotationWaiter.assertTrue(false);
+                }
+            }
+            catch (InterruptedException e) {
+                e.printStackTrace();
+            }
+
+            annotationWaiter.resume();
+        }).start();
+
+        // Wait for thread to complete
+        annotationWaiter.await();
+
+        String query = "<Named_entity.value=\"LOC\"/>";
+
+        ArrayList<SearchResult> results = (ArrayList<SearchResult>) searchService.query(user,
+                project, query);
+
+        // Test results
+
+        SearchResult expectedResult = new SearchResult();
+        expectedResult.setDocumentId(sourceDocument.getId());
+        expectedResult.setDocumentTitle("test");
+        expectedResult.setText("Galicia ");
+        expectedResult.setLeftContext("capital of ");
+        expectedResult.setRightContext("is ");
+        expectedResult.setOffsetStart(15);
+        expectedResult.setOffsetEnd(22);
+        expectedResult.setTokenStart(3);
+        expectedResult.setTokenLength(1);
+
+        ArrayList<SearchResult> expectedResults = new ArrayList<SearchResult>();
+        expectedResults.add(expectedResult);
+
+        assertNotNull(results);
+        if (results != null) {
+            assertEquals(1, results.size());
```

Comment:
Please use `AssertJ`

Reply:
Why? Do we have a standard for this?

Reply:
I think it reads better and you can describe what you are asserting. You also added it in the pom so I assumed you want to use it.
`assertThat(results).as("Results could be retrieved").hasSize(1)''

Reply:
The assert was in the pom probably due some copy-paste. Anyway, I like the idea of using the hasSize method, but I haven't been able to use it. The command that you've suggested gives me this syntax error: The method assertThat(T, Matcher<? super T>) in the type Assert is not applicable for the arguments (List<SearchResult>)